nlp/spacy_3_projects/scripts/visualize_data.py

```diff
@@ -0,0 +1,65 @@
+"""
+Visualize the data with Streamlit and spaCy.
+
+Based on: https://github.com/explosion/projects/blob/v3/tutorials/ner_drugs/scripts/visualize_data.py
+"""
+import streamlit as st
+import spacy
+from spacy import displacy
+from spacy.tokens import DocBin
+import srsly
+import typer
+
+
+@st.cache(allow_output_mutation=True)
+def load_data(filepath):
+    nlp = spacy.blank('de')
+    doc_bin = DocBin().from_disk(filepath)
+    docs = list(doc_bin.get_docs(nlp.vocab))
+
+    n_total_ents = 0
+    n_no_ents = 0
+    labels = set()
+
+    for doc in docs:
+        n_total_ents += len(doc.ents)
+        if not doc.ents:
+            n_no_ents += 1
+        labels.update([ent.label_ for ent in doc.ents])
+
+    return docs, labels, n_total_ents, n_no_ents
+
+
+def main(file_paths: str):
+    files = [p.strip() for p in file_paths.split(",")]
+    st.sidebar.title("Data visualizer")
+    st.sidebar.markdown(
+        "Visualize the annotations using [displaCy](https://spacy.io/usage/visualizers) "
+        "and view stats about the datasets. Showing only the first {} docs "
```

Comment:
Gven the `{}`, should this string be formatted here, or is this done elsewhere (I'm not familiar with the streamlit package)?

Reply:
No you're right. It should be an f string and should be {n_docs}. This line should then also be placed below line 42, where n_docs is defined.
Or you could just remove the "Showing only the first {} docs part ".